pkg/extender/extender.go

```diff
@@ -129,15 +117,14 @@ func (e *Extender) processFilterRequest(w http.ResponseWriter, req *http.Request
 
 	filteredNodes := []v1.Node{}
 	driverVolumes, err := e.Driver.GetPodVolumes(&pod.Spec, pod.Namespace)
-
 	if err != nil {
 		storklog.PodLog(pod).Warnf("Error getting volumes for Pod for driver: %v", err)
 		if _, ok := err.(*volume.ErrPVCPending); ok {
 			http.Error(w, "Waiting for PVC to be bound", http.StatusBadRequest)
 			return
 		}
 	} else if len(driverVolumes) > 0 {
-		driverNodes, err := e.Driver.GetNodes()
+		driverNodes, err := e.Driver.GetNodes(args.Nodes.Items)
```

Comment:
This still causes delays if only a small number of nodes are passed into the Filter request and there are a large number of nodes in the cluster.